src/Marten/Events/EventMapping.cs

```diff
@@ -159,6 +159,14 @@ public T Resolve(DbDataReader reader, IIdentityMap map)
             return map.Get<T>(id, json);
         }
 
+        public async Task<T> ResolveAsync(DbDataReader reader, IIdentityMap map, CancellationToken token)
+        {
+            var id = await reader.GetFieldValueAsync<Guid>(0, token).ConfigureAwait(false);
+            var json = await reader.GetFieldValueAsync<string>(1, token).ConfigureAwait(false);
```

Comment:
@jeremydmiller Can you tell me why event mapping has the fields reserved? So in the other classes the id is on index 1 and the json on index 0. See for example https://github.com/JasperFx/marten/pull/216/files#diff-bcb01cdb23b98acc4aefa1d606ef994bR76